src/templates/main.php

```diff
@@ -10,11 +10,11 @@
 <?php for ($i = 0; $i < count($lines); ++$i): ?>
     <path id='path<?php echo $i ?>'>
 <?php if (!$multiline): ?>
-        <animate id='d<?php echo $i ?>' attributeName='d' begin='<?php echo ($i == 0 ? "0s;" : "") . $previousId ?>.end' dur='5s'
+        <animate id='d<?php echo $i ?>' attributeName='d' begin='<?php echo ($i == 0 ? "0s;" : "") . $previousId ?>.end' dur='<?php echo 50 / $speed ?>s'
             values='m0,<?php echo $height / 2 ?> h0 ; m0,<?php echo $height / 2 ?> h<?php echo $width ?> ; m0,<?php echo $height / 2 ?> h0' keyTimes='0;0.8;1' />
 <?php else: ?>
     <?php $lineHeight = $size + 5;?>
-    <animate id='d<?php echo $i ?>' attributeName='d' dur='<?php echo 5 * ($i + 1) ?>s' fill="freeze"
+    <animate id='d<?php echo $i ?>' attributeName='d' dur='<?php echo 50 / $speed ?>s' fill="freeze"
```

Comment:
For multiline, the duration needs to multiplied by `$i + 1` since the delay of upcoming lines being displayed is included in the duration.
```suggestion
    <animate id='d<?php echo $i ?>' attributeName='d' dur='<?php echo (50 / $speed) * ($i + 1) ?>s' fill="freeze"
```